evaluation/metrics/metrics.py

```diff
@@ -20,6 +21,83 @@
 
 
 # eval_metrics, take predictions and targets as input
+def get_dataperf_accuracy(predictions: list, targets: list):
+    """
+    Here, p can be a list of acceptable multilabel lists, instead of just one multilabel
+    list. This is helpful for stochastic models, where we also want to report a
+    standard deviation over the model outputs.
+    """
+
+    if not isinstance(predictions[0], list):
+        predictions = [[pred] for pred in predictions]
+
+    tag_f1_scores = {}
+    tag_results = {}
+    for preds in predictions:
+        for pred in preds:
+            for tag in pred:
+                tag_results[tag] = {"t": [], "p": []}
+                tag_f1_scores[tag] = []
+    for tags in targets:
+        for tag in tags:
+            tag_results[tag] = {"t": [], "p": []}
+            tag_f1_scores[tag] = []
+
+    iterations = len(predictions[0])
+    mean_f1s = []
+    for iteration in range(iterations):
+        for tag in tag_results:
+            tag_results[tag] = {"t": [], "p": []}
+        for p, t in zip([pred[iteration] for pred in predictions], targets):
+            for tag in tag_results.keys():
+                if tag in p:
+                    tag_results[tag]["p"].append(1)
+                else:
+                    tag_results[tag]["p"].append(0)
+
+                if tag in t:
+                    tag_results[tag]["t"].append(1)
+                else:
+                    tag_results[tag]["t"].append(0)
```

Comment:
Just checking — for a given `tag`, the `p` list (predicted) will be something like [0, 1, 0...] and the `t` list (true) will be something like [0, 1, 1, 0] where predictions match up with targets if both are `1` or both are `0`.  So the `f1_score` being computed is the f1 score of a particular label being predicted (as in the f1_score of a binary classifier that only tells you whether a particular label should or should not be in the list of labels for an input)

Reply:
Yes, that's right. I'm going to check with the dataperf team to see if they want to modify this function.